core/sql/ustat/hs_parser.cpp

```diff
@@ -428,25 +435,22 @@ Lng32 AddTableName( const hs_table_type type
             LM->Log(LM->msg);
          }
 
-        const char* period = strchr(hs_globals->catSch->data(), '.');
-        NAString catName(hs_globals->catSch->data(),
-                         period - hs_globals->catSch->data());
+        NAString catName(hs_globals->objDef->getCatName());
+
+        *hs_globals->hstogram_table = getHistogramsTableLocation(hs_globals->catSch->data(), FALSE);
+
+        *hs_globals->hsintval_table = getHistogramsTableLocation(hs_globals->catSch->data(), FALSE);
```

Comment:
Looks like hsintval_table will get the same value as hstogram_table? (At least I can't see anything different on the right hand side of the two assigns)